airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/StreamingJdbcDatabase.java

```diff
@@ -105,11 +106,11 @@ public boolean tryAdvance(final Consumer<? super T> action) {
           return true;
         } catch (final SQLException e) {
           LOGGER.error("SQLState: {}, Message: {}", e.getSQLState(), e.getMessage());
-          streamException = e;
-          isStreamFailed = true;
-          // throwing an exception in tryAdvance() method lead to the endless loop in Spliterator and stream
-          // will never close
-          return false;
+          throw new RuntimeException(e);
```

Comment:
Is the comment "throwing an exception in tryAdvance() method lead to the endless loop" incorrect then?
This change is undoing the [fix](https://github.com/airbytehq/airbyte/pull/20888/files#diff-bb891ddb0da8855603fc22a299e26d4bb30fe1f492bfcda790a7d90eed90500e) we put in place to #10128.
Wouldn't that lead to connection leaks again?

Reply:
See https://github.com/airbytehq/airbyte/pull/27212
Have more notes here, but I've asserted that even if we throw an exception here, the `onClose` method is called and we attempt to close the connection. So, I'm not quite sure if the initial fix worked to begin with or was only tested on a particular conneciton